src/ad-services/live-connect/index.ts

```diff
@@ -47,57 +53,64 @@ export class LiveConnect extends BaseServiceSetup {
 				.loadScript(liveConnectScriptUrl, 'text/javascript', true, 'first')
 				.then(() => {
 					utils.logger(logGroup, 'loaded');
-					this.track();
+					this.resolveAndTrackIds();
 				});
 		} else {
 			communicationService.emit(eventsRepository.LIVE_CONNECT_CACHED);
 			utils.logger(logGroup, `already loaded and available in ${this.storageConfig.type}Storage`);
 		}
 	}
 
-	resolveId(idName, partnerName) {
-		return (nonId) => {
-			const partnerIdentityId = nonId[idName];
+	resolveAndTrackIds(): void {
+		if (!window.liQ) {
+			utils.warner(logGroup, 'window.liQ not available for tracking');
+			return;
+		}
+
+		window.liQ.resolve(
+			(result) => {
+				this.trackIds(result);
+			},
+			(err) => {
+				console.error(err);
+			},
+			{ qf: '0.3', resolve: ['sha1', 'sha2', 'md5', 'unifiedId'] },
+		);
+	}
+
+	trackIds(liQResponse: LiQResolveResponse): void {
+		utils.logger(logGroup, 'resolve response:', liQResponse);
 
-			utils.logger(logGroup, `id ${idName}: ${partnerIdentityId}`);
+		Object.keys(liQResponse).forEach((key) => {
+			const trackingKeyName = this.getTrackingKeyName(key);
+
+			if (!this.isAvailableInStorage(trackingKeyName)) {
```

Comment:
it makes more sense this way
```suggestion
			if (this.isAvailableInStorage(trackingKeyName)) {
```